src/game_objects/construct.cpp

```diff
@@ -53,19 +55,27 @@ Spikes::Spikes(int x, int z, int id) : Construct(x, z, id) {
         rigid.is_ghost = true;
         rigid.position = { x, 0.0f, z };
         events::add_rigidbody_event(rigid);
+        notify_on_collision = true;
     }
 }
 
 void Spikes::s_on_collision(GameObject *other) {
     Player *player = dynamic_cast<Player*>(other);
     if (player) {
-        events::dungeon::network_interact_event(other->get_id(), id);
+        events::dungeon::network_collision_event(other->get_id(), id);
+        if (player->s_take_damage()) {
+            // Send signal to client that this player was hit
+            events::dungeon::network_collision_event(id, other->get_id());
+        }
     }
 }
 
 void Spikes::c_on_collision(GameObject *other) {
-    anim_player.set_anim("trigger");
-    anim_player.play();
+    if (anim_player.check_paused()) {
```

Comment:
These lines (check_paused, set_anim, play) seem to be used a lot. Maybe we can extract these to a helper like `play_if_paused(string animation_name, float speed = 1.0f)`.